src/main/java/io/confluent/avro/random/generator/Generator.java

```diff
@@ -352,6 +356,21 @@ private Object generateObject(Schema schema) {
     }
   }
 
+  private <L extends LogicalType> L getLogicalType(
+      Schema schema,
+      String logicalTypeName,
+      Class<L> logicalTypeClass
+  ) {
+    return Optional.ofNullable(schema.getLogicalType())
+        .filter(logicalType -> Objects.equals(logicalTypeName, logicalType.getName()))
+        .map(logicalTypeClass::cast)
+        .orElse(null);
+  }
+
+  private LogicalTypes.Decimal getDecimalLogicalType(Schema schema) {
+    return getLogicalType(schema, "decimal", LogicalTypes.Decimal.class);
```

Comment:
Our schemas use the [`Decimal.LOGICAL_NAME`](https://github.com/apache/kafka/blob/trunk/connect/api/src/main/java/org/apache/kafka/connect/data/Decimal.java#L41) constant to detect these logical types. For example, see the [Jdbc Connector](https://github.com/confluentinc/kafka-connect-jdbc/blob/master/src/main/java/io/confluent/connect/jdbc/dialect/MySqlDatabaseDialect.java#L87-L95)
```suggestion
    return getLogicalType(schema, Decimal.LOGICAL_NAME, LogicalTypes.Decimal.class);
```
I think as implemented, logical decimals produced wouldn't be correctly recognized by downstream systems.

I understand that this would be an additional dependency for this project, so could you describe how we might make changes to the datagen connector to make these two compatible? Without any changes, this is a potential footgun when someone tries to use them together and it gets written as raw bytes.

Otherwise, could you pull this out to a new or existing constant? Maybe even `LogicalTypes.Decimal.class.getName()`

Reply:
> I think as implemented, logical decimals produced wouldn't be correctly recognized by downstream systems.

The `AvroData` class has [logic for preserving logical types](https://github.com/confluentinc/schema-registry/blob/3b978da07cb83724430cf675d226f26333408110/avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java#L1566-L1586) when converting schemas from Avro to Connect, so it should be fine to stick to the logical type name defined in the Avro spec. This project isn't (currently) aware of Connect schema formats, so I'd be very hesitant to add Connect-specific support, especially since the Avro converter is already capable of taking in a purely Avro schema and preserving its logical type information.

There wouldn't be any changes necessary to the Datagen connector (except maybe a bump in version for the dependency that draws in the `AvroData` class? Don't know when logical type support for `Decimal` was added) since, assuming it were using a version of the generator that includes the feature implemented in this PR, it would already support Avro schemas with the `Decimal` logical type.

Unfortunately, afaict the Avro Java library doesn't actually define public constants for the names of its logical types. However, the Avro spec is pretty clear that it should just be `"decimal"`. I'd rather not use reflection or class names for something so simple, but it wouldn't hurt to define a new string constant for the expected logical type name for Avro `Decimal`s.

Reply:
SGTM, thanks for clarifying. I didn't know that AvroData is so flexible :)